index.js

```diff
@@ -240,8 +246,35 @@ function constructRequest(har, opts = { userAgent: false, files: {}, multipartEn
 
           options.body = JSON.stringify(formBody);
       }
-    } else {
-      options.body = request.postData.text;
+    } else if (request.postData.text.length) {
+      // If we've got `files` map content present, and this post data content contains a valid data URL then we can
+      // substitute the payload with that file instead of the using data URL.
+      if (opts.files) {
+        const parsed = parseDataUrl(request.postData.text);
+        if (parsed && 'name' in parsed && parsed.name in opts.files) {
+          const fileContents = opts.files[parsed.name];
+          if (Buffer.isBuffer(fileContents)) {
```

Comment:
How does the browser react to `Buffer`?

Reply:
It wasn't throwing any exceptions in browsers, though it probably should, so I've changed these over to an `isBuffer()` function call that does a `typeof Buffer !== 'undefined'` check before hitting `Buffer.isBuffer()`.